### ISSUE TITLE
az ssh config --file crashes if given filename with no /

### ISSUE THREAD

**Issue by roy-work** [Azure/azure-cli-extensions#3512]
### Extension name (the extension in question)
The `az ssh` extension

### Description of issue (in as much detail as possible)
Attempting to run,
```
» az ssh config -f foo -g test -n test-new-aad-ext
```
Results in the following crash:
```
CLIInternalError: The command failed with an unexpected error. Here is the traceback:
[Errno 2] No such file or directory: ''
Traceback (most recent call last):
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/knack/cli.py", line 215, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 654, in execute
    raise ex
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 717, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 710, in _run_job
    six.reraise(*sys.exc_info())
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/six.py", line 703, in reraise
    raise value
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 688, in _run_job
    result = cmd_copy(params)
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/azure/cli/core/commands/__init__.py", line 325, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/local/Cellar/azure-cli/2.16.0/libexec/lib/python3.8/site-packages/azure/cli/core/__init__.py", line 784, in default_command_handler
    return op(**command_args)
  File "/Users/roy/.azure/cliextensions/ssh/azext_ssh/custom.py", line 28, in ssh_config
    _do_ssh_op(cmd, resource_group_name, vm_name, ssh_ip, public_key_file, private_key_file, use_private_ip, op_call)
  File "/Users/roy/.azure/cliextensions/ssh/azext_ssh/custom.py", line 49, in _do_ssh_op
    op_call(ssh_ip, username, cert_file, private_key_file)
  File "/Users/roy/.azure/cliextensions/ssh/azext_ssh/ssh_utils.py", line 54, in write_ssh_config
    file_utils.make_dirs_for_file(config_path)
  File "/Users/roy/.azure/cliextensions/ssh/azext_ssh/file_utils.py", line 12, in make_dirs_for_file
    mkdir_p(os.path.dirname(file_path))
  File "/Users/roy/.azure/cliextensions/ssh/azext_ssh/file_utils.py", line 17, in mkdir_p
    os.makedirs(path)
  File "/usr/local/Cellar/python@3.8/3.8.6_2/Frameworks/Python.framework/Versions/3.8/lib/python3.8/os.py", line 223, in makedirs
    mkdir(name, mode)
FileNotFoundError: [Errno 2] No such file or directory: ''
To open an issue, please run: 'az feedback'
```

This is because `write_ssh_config` attempts to make any non-existing directories: https://github.com/Azure/azure-cli-extensions/blob/568b8a889d99c3d1ff8ea67a9ce2925ccb8aeaf1/src/ssh/azext_ssh/ssh_utils.py#L52-L54

Which calls this:

https://github.com/Azure/azure-cli-extensions/blob/568b8a889d99c3d1ff8ea67a9ce2925ccb8aeaf1/src/ssh/azext_ssh/file_utils.py#L10-L12

The immediate problem is that, for `foo`, `os.path.dirname` returns `''` ; it then tries to create this directory, which of course fails.

While that's the immediate cause, I would really not expect this utility to be creating directories like that. It's only being asked to emit the config to the path that I've specified. If a directory in that path doesn't exist, just error out & tell me. Worse, if I typo, it will created the typo'd directory. The examples in the docs for this flag also use `~/.ssh`, which needs to have strict permissions on it, which this would ignore. It's better to just to what was asked (emit the config) & not try to correct for mistakes here.

**Comment by yonzhan** [comment]
VM SSH

**Comment by N6UDP** [comment]
Fixed in #3698

**Comment by maertendMSFT** [comment]
closing as it is called out as fixed.
